core/src/main/java/de/danoeh/antennapod/core/feed/TagFilter.java

```diff
@@ -0,0 +1,50 @@
+package de.danoeh.antennapod.core.feed;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import de.danoeh.antennapod.core.storage.NavDrawerData;
+import de.danoeh.antennapod.model.feed.FeedPreferences;
+
+public class TagFilter {
```

Comment:
Hmm, this seems rather strange to me. It is basically a duplicate of SubscriptionsFilter. Instead, I would suggest to filter the subscriptions and then get their tags.